lib/galaxy/tool_util/deps/container_resolvers/mulled.py

```diff
@@ -460,7 +460,7 @@ def resolve(self, enabled_container_types, tool_info, install=False, session=Non
                 shell=self.shell,
             )
             if self.can_list_containers:
-                if install and not self.cached_container_description(
+                if (install or self.auto_install) and not self.cached_container_description(
```

Comment:
Ah, I think I remember now why we did not look at self.auto_install here. We don't want to pull containers when we list them in the admin panel. @bernt-matthias can you verify that just listing the containers in the admin panel doesn't list them ?

Reply:
I tried on a local instance since our cluster does not support docker. 

In the default config, listing in the admin panel pulls the containers. 

They are not pulled if I set `containers_resolvers_config_file: container_resolvers_conf.xml` in `galaxy.yml` and 

use the following config:
```
<containers_resolvers>
  <explicit auto_install="False"/>
  <cached_mulled auto_install="False"/>
  <mulled auto_install="False"/>
  <build_mulled auto_install="False"/>
</containers_resolvers>
```

I tried to change the default resolvers here https://github.com/galaxyproject/galaxy/blob/c3fc59114842ddc78d3dcd3929c003b232c80486/lib/galaxy/tool_util/deps/containers.py#L219 but this led to a loop in listing the containers. Also the init methods do not pass auto_install to the constructors of the super classes .. which seems odd .. but I did not have the time to think about the semantics of these classes and their relationships. 

Is the default `auto_install=True` desired for some reason? 

If I get the docs in the sample config right, then `auto_install` should control if the container is downloaded at tool run. But it seems that it also triggers installation at listing which seems totally undesired to me. Is there a way to determine if the resolve function is called during listing / tool run?

Anyway we should have it consistent over all resolvers, or: https://github.com/galaxyproject/galaxy/blob/8b7609f9a7c4588c24009eac755c7e27b049e803/lib/galaxy/tool_util/deps/container_resolvers/mulled.py#L553?